samcli/local/docker/container.py

```diff
@@ -180,6 +180,34 @@ def create(self):
 
         return self.id
 
+    def stop(self, time=3):
+        """
+        Stop a container, with a given number of seconds between sending SIGTERM and SIGKILL.
+
+        Parameters
+        ----------
+        time
+            Optional. Number of seconds between SIGTERM and SIGKILL. Effectively, the amount of time
+            the container has to perform shutdown steps. Default: 3
+        """
+        if not self.is_created():
+            LOG.debug("Container was not created, cannot run stop.")
+            return
+
+        try:
+            self.docker_client.containers.get(self.id).stop(timeout=time)
+        except docker.errors.NotFound:
+            # Container is already removed
+            LOG.debug("Container with ID %s does not exist. Cannot stop!", self.id)
+        except docker.errors.APIError as ex:
+            msg = str(ex)
```

Comment:
Could you use `ex.message` instead of this? (does it contain both text parts you are searching?)

Reply:
We already use this logic as-is in `delete` which works, so for this one inclined to not mess with it.